ci/windows_ci_steps.sh

```diff
@@ -20,14 +20,22 @@ df -h
 # rules_foreign_cc does not currently use bazel output/temp directories by default, it uses mktemp
 # which respects the value of the TMPDIR environment variable
 drive="$(readlink -f $TMPDIR | cut -d '/' -f2)"
-/c/windows/system32/cmd.exe "/c mklink /d $drive:\\$drive $drive:\\"
+if [ ! -e "/$drive/$drive" ]; then
+  /c/windows/system32/cmd.exe /c "mklink /d $drive:\\$drive $drive:\\"
+fi
 
 BAZEL_STARTUP_OPTIONS="--noworkspace_rc --bazelrc=windows/.bazelrc --output_base=c:/_eb"
 BAZEL_BUILD_OPTIONS="-c opt --config=msvc-cl --show_task_finish --verbose_failures \
   --test_output=all ${BAZEL_BUILD_EXTRA_OPTIONS} ${BAZEL_EXTRA_TEST_OPTIONS}"
 
-bazel ${BAZEL_STARTUP_OPTIONS} build ${BAZEL_BUILD_OPTIONS} //bazel/... --build_tag_filters=-skip_on_windows
+# With all envoy-static and //test/ tree building, no need to test compile externals
+# bazel ${BAZEL_STARTUP_OPTIONS} build ${BAZEL_BUILD_OPTIONS} //bazel/... --build_tag_filters=-skip_on_windows
 
 bazel ${BAZEL_STARTUP_OPTIONS} build ${BAZEL_BUILD_OPTIONS} //source/exe:envoy-static --build_tag_filters=-skip_on_windows
 
-# bazel ${BAZEL_STARTUP_OPTIONS} test ${BAZEL_BUILD_OPTIONS} //test/... --test_tag_filters=-skip_on_windows --build_tests_only --test_summary=terse --test_output=errors
+# Test compilation of known MSVC-compatible test sources
+bazel ${BAZEL_STARTUP_OPTIONS} build ${BAZEL_BUILD_OPTIONS} //test/... --test_tag_filters=-skip_on_windows --build_tests_only
```

Comment:
Why just build but not run?

Reply:
This change was just an effort to ensure compilation of test sources is checked in CI. Not all the tests pass yet, we have not yet started delving into the test failures/figuring out what is flaky/fixing them fully. We didn't want to run all tests, have them fail intermittently, and break other PRs/pushes. If desired, we can tag all the tests that fail with `fails_on_windows` (no guarantees we find all the flaky ones).

Reply:
Right now, about 73 tests fail. After the next follow-up PR, we expect that about 5 corrections and abstractions will resolve 80% of those failures. Trying to keep the scope of each PR review reasonably manageable.

We've been increasing the stress on the wider community slowly. Thus far, we are simply trying to ensure new compilation breakage isn't introduced due to clang/gcc assumptions that MSVC doesn't interpret in the same way, and have been assisting PR contributors with guidance in resolving these compilation discrepancies for envoy-static. We'll continue to do so with the newly introduced //test/... compilation tests.

The resulting behavior differences of the Windows architecture begin to be addressed in next major PR, with a basket of remaining broken tests marked fails_on_windows on a short-term basis, to keep focus on not introducing regressions. The remaining 20% of failures will each demand more detailed fixes (likely 1 PR per failure) and closer scrutiny.

The end goal is 100% success and relies on additional volunteers (outside of our company) who are joining the effort already.